src/dotnet/Chat.UI.Blazor/Components/ChatView/ChatView.razor.cs

```diff
@@ -233,4 +229,17 @@ private Task OnNavigateToChatEntry(NavigateToChatEntry navigation, CancellationT
         NavigateToEntry(navigation.ChatEntryId);
         return Task.CompletedTask;
     }
+
+    private void OnLocationChanged(object? sender, LocationChangedEventArgs e)
+        => TryNavigateToEntryIfExists();
```

Comment:
OnLocationChanged is invoked once even after dispose was called when you are navigating out of chat page. I think make sense to set isDisposed flag in Dispose method and check it here. I will allow to avoid situation when we are navigating out of chat page to url with hash and it spawn avigating to chat entry on disposed component.